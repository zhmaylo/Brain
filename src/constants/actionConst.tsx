/////////////////////////////
//Start. Search section
//
// search string
export const  SEARCH_REQUEST = 'SEARCH_REQUEST';
// 
//End. Search section
/////////////////////////////

/////////////////////////////
//Start. Categories list section
//
// categories list (JSON format)
export const  CATEGORY_LIST = 'CATEGORY_LIST';
//
// view categories list (JSON format)
export const  VIEW_LIST_CAT  = 'VIEW_LIST_CAT';
//
// 
//End. Categories list section
/////////////////////////////


/////////////////////////////
//Start. Filter section
//
// filter all 
export const  MINMAX_DEAL_PRICE = 'MINMAX_DEAL_PRICE';
export const  MIN_SHOW_LIMIT = 'MIN_SHOW_LIMIT';
export const  MAX_SHOW_LIMIT = 'MAX_SHOW_LIMIT';
// 
//End. Filter section
/////////////////////////////


/////////////////////////////
//Start. App init section
//
// application initialization flag
export const IS_APP_INIT = 'IS_APP_INIT';
//
//End. App init section
/////////////////////////////


/////////////////////////////
//Start. Login section
//
// login
export const  LOGIN = 'LOGIN';
// password
export const  PASSWORD = 'PASSWORD';
// 
// passwordMD5
export const  PASS_MD5 = 'PASS_MD5';
// login error
export const  LOGIN_ERROR = 'LOGIN_ERROR';
// 
//End. Login section
/////////////////////////////


/////////////////////////////
//Start. Number column of main screen section
//
// number of columns of displayed products
export const  NUM_COLUM_PROD = 'NUM_COLUM_PROD';
//
//End. Number column of main screen section
/////////////////////////////


/////////////////////////////
//Start. Product current section
//
// product current description
export const  PROD_CURR_DESCRIPTION = 'PROD_CURR_DESCRIPTION';
//
// product current images
export const  PROD_CURR_IMAGES = 'PROD_CURR_IMAGES';
//
//End. Product current section
/////////////////////////////


/////////////////////////////
//Start. Product list section
//
// products list (JSON format)
export const  PRODUCTS_LIST = 'PRODUCTS_LIST';
//
// size view list products
export const  SIZE_LIST_PROD = 'SIZE_LIST_PROD';
//
//End. Product list section
/////////////////////////////


/////////////////////////////
//Start. Progress Bar section
//
// category loading level (for progress bar)
export const  PROG_BAR_MAX_VOL = 'PROG_BAR_MAX_VOL';
export const  PROG_BAR_CUR_VOL = 'PROG_BAR_CUR_VOL';
// 
//End. Progress Bar section
/////////////////////////////


/////////////////////////////
//Start. SID section
//
// SID session 
export const  SESSION_SID  = 'SESSION_SID';
//
//End. SID section
/////////////////////////////


/////////////////////////////
//Start. Sort list products section
//
// ascending sort by price
export const  SORT_PRICE_ASCEND = 'SORT_PRICE_ASCEND';
// descending sort by price
export const  SORT_PRICE_DESCEND = 'SORT_PRICE_DESCEND';
//
// ascending sort by vendor
export const  SORT_VENDOR_ASCEND = 'SORT_VENDOR_ASCEND';
// descending sort by vendor
export const  SORT_VENDOR_DESCEND = 'SORT_VENDOR_DESCEND';
//
// ascending sort by warranty
export const  SORT_WARRANTY_ASCEND = 'SORT_WARRANTY_ASCEND';
// descending sort by warranty
export const  SORT_WARRANTY_DESCEND = 'SORT_WARRANTY_DESCEND';
//
//End. Sort list products section
/////////////////////////////


/////////////////////////////
//Start. Spiner section
//
// spiner toggle (off(false)/ on(true))
export const SPINER_TOGGLE = 'SPINER_TOGGLE';
// 
//End. Spiner section
/////////////////////////////


/////////////////////////////
//Start. Status response section
//
// status response (error/ok)
export const  STATUS_RESPONSE = 'STATUS_RESPONSE';
//
//End. Status response section
/////////////////////////////


/////////////////////////////
//Start. SyncData section
//
// DB work log
export const  DB_LOG = 'DB_LOG';
// 
//End. SyncData section
/////////////////////////////


/////////////////////////////
//Start.  section
//
//
//End.  section
/////////////////////////////


/////////////////////////////
//Start.  section
//
//
//End.  section
/////////////////////////////


/////////////////////////////
//Start.  section
//
//
//End.  section
/////////////////////////////


