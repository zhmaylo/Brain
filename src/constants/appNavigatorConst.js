
/////////////////////////////
// Start. Section 'menu - string values of items'
//
export const MAIN_SCR = "MainScreen";
export const CATEG_SCR = "CategoryScreen";
export const SORT_SCR = "SortScreen";
export const PROD_SCR = "ProdScreen";
export const PROD_GALL_SCR = "ProdGalleryScreen";
export const SYNC_DATA_SCR = "SyncDataScreen";
//
// End. Section 'menu - string values of items'
/////////////////////////////





