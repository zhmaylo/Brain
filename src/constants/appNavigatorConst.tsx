
/////////////////////////////
// Start. Section 'menu - string values of items'
//
export const MAIN_SCR = "MainScreen";
export const CATEG_SCR = "CategoryScreen";
export const SORT_SCR = "SortScreen";
export const PROD_SCR = "ProdScreen";
export const FILTER_SCR = "FilterScreen";
export const SEARCH_SCR = "SearchScreen";
export const PROD_GALL_SCR = "ProdGalleryScreen";
export const SYNC_DATA_SCR = "SyncDataScreen";
export const SETTING_SCR = "SettingScreen";
export const LOGIN_SCR = "LoginScreen";
//
// End. Section 'menu - string values of items'
/////////////////////////////





