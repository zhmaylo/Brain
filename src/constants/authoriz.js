
/////////////////////////////
// Start. Section 'User authorization data'
//
// API_KEY
export const API_KEY = "7408fbfbded5974d9b625d05237f6a6f";
//
// user login/password
export const LOGIN = "dzhmaylo@gmail.com";
export const PASSMD5 = "b4a62dea0279a07036a87e355af638b4";
export const USER_AUTH = 'login=' + LOGIN + '&password=' + PASSMD5;
//
// End. Section 'User authorization data'
/////////////////////////////

