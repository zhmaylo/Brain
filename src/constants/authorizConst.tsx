
/////////////////////////////
// Start. Section 'User authorization data'
//
// 
// 
// user login/password
export const LOGIN_DEFAULT = "dzhmaylo@gmail.com";
let login: string = LOGIN_DEFAULT;
export const setLogin = (loginFromStore: string) => {
    login = loginFromStore;
};

export const PASS_MD5_DEFAULT = "b4a62dea0279a07036a87e355af638b4";
let passMD5: string = PASS_MD5_DEFAULT;
export const setPassMD5 = (PassFromStore: string) => { passMD5 = PassFromStore };

export const USER_AUTH = () => 'login=' + login + '&password=' + passMD5;

//request header for authorization
export const REQUEST_HEADER_AUTH = () => [{
    method: 'POST',
    headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
    body: USER_AUTH()
}];
//
// End. Section 'User authorization data'
/////////////////////////////


/////////////////////////////
// Start. Section 'Time Stamp'
//
// time life session - 5 min = 300000 ms
// export const SID_TIME_LIFE = 300000;
//
// End. Section 'Time Stamp'
/////////////////////////////