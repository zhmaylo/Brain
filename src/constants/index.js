//constants for actions

export const  LOAD_DATA_JSON  = 'LOAD_DATA_JSON'
export const  LOAD_JSON_FLAG  = 'LOAD_JSON_FLAG'
export const  STORAGE_TO_STATE  = 'STORAGE_TO_STATE'



