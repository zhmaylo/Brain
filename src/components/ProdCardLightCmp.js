import React from 'react';
import { View, StyleSheet, Text, Image } from 'react-native';
import { WINDOW_WIDTH, WINDOW_HEIGHT } from '../constants/otherConst';
import { WARANTY_HOME, WARANTY_END, ID_HOME } from '../constants/productsConst';


// product card
export const ProdCardLightCmp = ({ item, numCol }) => {
    // console.log("ProductsCardCmp=>item", item);
    // console.log("ProductsCardCmp=>item.small_image", item.small_image);
    return (
        <View width={WINDOW_WIDTH / numCol} style={styles.itemProd} >
            <Image
                style={styles.image}
                source={{ uri: (item.large_image) }}
                // source={require('./../icons/basket.png')}
            />
            <Text style={styles.textName}>{item.name} </Text>
            <Text style={styles.textPrice}>{item.retail_price_uah} грн </Text>
            <Text style={styles.textName}>{WARANTY_HOME}{item.warranty}{WARANTY_END} {ID_HOME}{item.productID} </Text>
        </View>
    )
}


const styles = StyleSheet.create({
    image: {

        height: WINDOW_HEIGHT / 3.0,
        resizeMode: "contain",

    },

    itemProd: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'space-around',
        backgroundColor: "white",
        margin: 1,
        padding: 5,


    },
    textName: {
        fontSize: 16,
        paddingHorizontal: 5,
    },

    textPrice: {
        fontSize: 16,
        fontWeight: "700",
        paddingHorizontal: 5,
    }
})
