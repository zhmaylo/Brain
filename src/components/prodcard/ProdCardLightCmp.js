import React from 'react';
import { View, StyleSheet, Text, Image, TouchableOpacity } from 'react-native';
import { WINDOW_HEIGHT } from '../../constants/otherConst';
import { WARANTY_HOME, WARANTY_END, ID_HOME, CODE_HOME } from '../../constants/productsConst';
import { PROD_SCR } from '../../constants/appNavigatorConst';
import { PROD_CURR_DESCRIPTION, PROD_CURR_IMAGES } from '../../reducers/prodCurrentRdc';
import { prodImgLoad } from '../../api/products/prodImgLoad';
import { SPINER_TOGGLE } from '../../reducers/spinerRdc';

// product card
export const ProdCardLightCmp = ({ item, props, dispatch, sidAndTime }) => {
    // console.log("ProductsCardLightCmp=>props", props);
    // console.log("ProductsCardCmp=>item", item);
    // console.log("ProductsCardCmp=>item.small_image", item.small_image);
    return (
        <View style={styles.itemProd}>
            <TouchableOpacity
                onPress={() => {
                    dispatch({ type: PROD_CURR_DESCRIPTION, payload: item });
                    console.log('ProdCardLightCmp.item=> ', item);
                    prodImgLoad(item.productID, sidAndTime, dispatch).then(currImages => {
                        dispatch({ type: PROD_CURR_IMAGES, payload: currImages });
                        console.log('ProdCardLightCmp.currImg=> ', currImages);
                    })
                    props.navigation.navigate(PROD_SCR);
                
                }}
            >
                <View>
                    <Image
                        style={styles.image}
                        source={{ uri: (item.large_image) }}
                    />
                    <Text style={styles.textName}>{item.name} </Text>
                    <Text style={styles.textPrice}>{item.retail_price_uah} грн </Text>
                    <Text style={styles.text_warranty}>{WARANTY_HOME}{item.warranty}{WARANTY_END} {CODE_HOME}{item.product_code} </Text>
                    <Text style={styles.text_warranty}>{ID_HOME}{item.productID} </Text>
                </View>
            </TouchableOpacity>
        </View>
    )
}


const styles = StyleSheet.create({
    itemProd: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'space-around',
        backgroundColor: "white",
        margin: 1,
        padding: 5,
    },

    image: {
        height: WINDOW_HEIGHT / 5,
        resizeMode: "contain",
    },

    textName: {
        fontSize: 14,
        paddingHorizontal: 5,
    },

    text_warranty: {
        fontSize: 12,
        paddingHorizontal: 5,
    },

    textPrice: {
        fontSize: 16,
        fontWeight: "700",
        paddingHorizontal: 5,
    }
})
