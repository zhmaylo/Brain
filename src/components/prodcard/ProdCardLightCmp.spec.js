
import renderer from 'react-test-renderer';
import { PROD_SCR } from '../../constants/appNavigatorConst';
import { PRODUCTS_FROM_FILE } from '../../constants/productsJSON';
import { ProdCardLightCmp } from './ProdCardLightCmp';


let item = PRODUCTS_FROM_FILE[0].result.list;
let sidAndTime = {
    sid: "m8a0ja0dbelrdbaossblnl4a40",
    timeStamp: 1619522162997
}


test('renders correctly', () => {
    const navigate = () => {
        return true;
    };
    let navigation = { navigate };
    let props = { navigation };

    const tree = renderer.create(


        ProdCardLightCmp({
            item: item,
            props: props,
            dispatch: null,
            sidAndTime: sidAndTime
        })
    )
        .toJSON();
    expect(tree).toMatchSnapshot();
});



// tasca - testAsyncCallback Creates Async Callback test
// taspr - testAsyncPromise Creates Aysnc Promise test
// tasaw - testAsyncAwait Creates Async Await test
// tsnsh - testSnapShot Creates Snapshot test
// tshre - testShallowRenderer Creates Shallow Renderer test