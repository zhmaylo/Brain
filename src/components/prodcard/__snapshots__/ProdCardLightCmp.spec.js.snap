// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-around",
      "margin": 1,
      "padding": 5,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View>
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": 266.8,
            "resizeMode": "contain",
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 14,
            "paddingHorizontal": 5,
          }
        }
      >
         
      </Text>
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "700",
            "paddingHorizontal": 5,
          }
        }
      >
        undefined / undefined
         грн 
      </Text>
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "700",
            "paddingHorizontal": 5,
          }
        }
      >
        NaN
         % / 
        NaN
         грн
      </Text>
      <Text
        style={
          Object {
            "fontSize": 12,
            "paddingHorizontal": 5,
          }
        }
      >
        Гарантия: 
        мec.
         
        Код: 
         
        ID: 
      </Text>
      <Text
        style={
          Object {
            "fontSize": 12,
            "paddingHorizontal": 5,
          }
        }
      />
    </View>
  </View>
</View>
`;
